packages/vite/src/node/server/hmr.ts

```diff
@@ -467,6 +467,7 @@ async function readModifiedFile(file: string): Promise<string> {
 async function restartServer(server: ViteDevServer) {
   // @ts-ignore
   global.__vite_start_time = Date.now()
+  const prevAddress = server.httpServer ? server.httpServer.address() : null
```

Comment:
```suggestion
  const port = server.httpServer?.address()?.port
```

Reply:
https://nodejs.org/api/net.html#net_server_address
> For a server listening on a pipe or Unix domain socket, the name is returned as a string.

`address` may be a string, so ts will throw an error, should I force the declaration of `address()` as `AddressInfo`?

![image](https://user-images.githubusercontent.com/15319872/127864240-900b8752-c56d-499d-afa1-35b69105d5b1.png)

Reply:
When would it be a string tho? If so, I think we'd better to store the resolved port somewhere instead of relying on this api.